### PR TITLE
Fix discover granules to support http/https protocols

### PR DESCRIPTION
**Pull request by amarouane-ABDLHAK**

### CODE REVIEW

File: packages/ingest/granule.js [hunk 1]
```diff
@@ -56,13 +56,17 @@ class Discover {
     const file = _file;
     let test = new RegExp(this.collection.granuleIdExtraction);
     const match = file.name.match(test);
+    //console.log(match);
     if (match) {
-      granuleId = match[1];
+    
+      granuleId = match[0];
```
Comment (scisco) [inline packages/ingest/granule.js]: @amarouane-ABDLHAK curious, why did you change this to `match[0]`

File: packages/ingest/granule.js [hunk 2]
```diff
@@ -85,25 +89,39 @@ class Discover {
...
+      //console.log(file);
       if (file) updatedFiles.push(file);
     });
+    //console.log(updatedFiles);    
     return await this.findNewGranules(updatedFiles);
   }
 
   async fileIsNew(file) {
+  // If the bucket is defined 
+  if(file.bucket){
```
Comment (scisco) [inline packages/ingest/granule.js]: @amarouane-ABDLHAK what are the cases where the file is missing a bucket?

**Comment by scisco** [comment]
@amarouane-ABDLHAK we had to make a few changes to your PR before we can merge it to master. Will track the PR here #116

File: packages/ingest/granule.js (hunk 1 above)
Comment (amarouane-ABDLHAK) [reply packages/ingest/granule.js]: We want the whole match of the file name to be the granuleId. match[1] will not work with a generic Regex expression for example if my regex is `^ghrc.*$` (all the files that start with keyword ghrc) and let say filename = 'ghrctest1.nc'
let test = new RegExp(`^ghrc.*$`);
match = filename.match(test); 
match[1] will be undefined.
image

Comment (scisco) [reply packages/ingest/granule.js]: Your extraction regex must include a capturing group for this work: https://www.regular-expressions.info/refcapture.html

File: packages/ingest/granule.js (hunk 2 above)
Comment (amarouane-ABDLHAK) [reply packages/ingest/granule.js]: It could happen if the granuleIdExtraction regex matches the provided file but it does not match any regex within collection.meta.files

Comment (scisco) [reply packages/ingest/granule.js]: I thought files that are not matched in collection.meta.files are ignored, that's not the case?

Comment (amarouane-ABDLHAK) [reply packages/ingest/granule.js]: I thought that too but I got some undefined bucket :) and digging into the source code I saw that the files not matching the regex but matching the granuleIdExtraction got the undefined buckets.

File: packages/ingest/granule.js (hunk 1 above)
Comment (amarouane-ABDLHAK) [reply packages/ingest/granule.js]: Yes even with capturing the group match[0] will still give you the filename as the granuleId and prevent an error where somebody did not use include the capturing group in the regex. But I am glad it was fixed :)

File: packages/ingest/granule.js (hunk 2 above)
Comment (scisco) [reply packages/ingest/granule.js]: interesting. This is certainly something we have to fix.

Comment (scisco) [reply packages/ingest/granule.js]: Filed a ticket for it: https://bugs.earthdata.nasa.gov/browse/CUMULUS-261
